shared/chat/conversation/info-panel/panels.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react'
+import * as Types from '../../../constants/types/chat2'
+import * as Kb from '../../../common-adapters'
+import * as Styles from '../../../styles'
+import RetentionPicker from '../../../teams/team/settings-tab/retention/container'
+import MinWriterRole from './min-writer-role/container'
+import Notifications from './notifications/container'
+import {CaptionedDangerIcon} from './channel-utils'
+
+type SettingsPanelProps = {
+  canDeleteHistory: boolean
+  conversationIDKey: Types.ConversationIDToKey
+  entityType: 'adhoc' | 'channel' | 'small team'
+  ignored: boolean
+  onHideConv: () => void
+  onUnhideConv: () => void
+  onLeaveConversation: () => void
+  onShowBlockConversationDialog: () => void
+  onShowClearConversationDialog: () => void
+  spinnerForHide: boolean
+  teamname: string
+  channelname?: string
+}
+
+export const SettingsPanel = (props: SettingsPanelProps) => {
+  return (
+    <Kb.Box2 direction="vertical" fullWidth={true} fullHeight={true} style={styles.settingsContainer}>
+      <Kb.ScrollView>
+        <Notifications conversationIDKey={props.conversationIDKey} />
+        <Kb.Divider style={styles.divider} />
+        <RetentionPicker
+          containerStyle={styles.retentionContainerStyle}
+          conversationIDKey={
+            ['adhoc', 'channel'].includes(props.entityType) ? props.conversationIDKey : undefined
+          }
+          dropdownStyle={styles.retentionDropdownStyle}
+          entityType={props.entityType}
+          showSaveIndicator={true}
+          teamname={props.teamname}
+          type="auto"
+        />
+        {(props.entityType === 'channel' || props.entityType === 'small team') && (
+          <>
+            <Kb.Divider style={styles.divider} />
+            <MinWriterRole
+              conversationIDKey={props.conversationIDKey}
+              isSmallTeam={props.entityType === 'small team'}
+            />
+          </>
+        )}
+        <Kb.Divider style={styles.divider} />
+        {(props.canDeleteHistory || props.entityType === 'adhoc') && (
```

Comment:
Shouldn't canDeleteHistory always be true if entityType === 'adhoc' ? If not, why?

Reply:
Fair enough, will change